### PR TITLE
[Feature][Datasource]replace DataSourceController API with string JSON and cast to DTO

### PR DESCRIPTION
**Pull request by Tianqi-Dotes**
## Purpose of the pull request

replace DataSourceController API with string JSON and cast to DTO

## Brief change log

replace DataSourceController API with string JSON and cast to DTO

## Verify this pull request

test locally

### CODE REVIEW

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/ApiApplicationServer.java [hunk 1]
```diff
@@ -43,5 +43,7 @@ public static void main(String[] args) {
     public void run(ApplicationReadyEvent readyEvent) {
         // install task plugin
         taskPluginManager.installPlugin();
+        DataSourceUtils dataSourceUtils = new DataSourceUtils();
```
Comment (caishunfeng) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/ApiApplicationServer.java]: It‘s better to use sprint bean, and maybe `DataSourceManager` is better, WDYT?

File: dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-api/src/main/java/org/apache/dolphinscheduler/plugin/datasource/api/utils/DataSourceUtils.java [hunk 2]
```diff
@@ -17,43 +17,55 @@
...
-    private static final DataSourceProcessor redshiftProcessor = new RedshiftDataSourceProcessor();
+    private static final Map<String, DataSourceProcessor> dataSourceProcessorMap = new ConcurrentHashMap<>();
+
+    public void installProcessor() {
+        final Set<String> names = new HashSet<>();
+
+        ServiceLoader.load(DataSourceProcessor.class).forEach(factory -> {
+            final String name = factory.getDbType().name();
+
+            logger.info("start register processor: " + name);
```
Comment (caishunfeng) [inline dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-api/src/main/java/org/apache/dolphinscheduler/plugin/datasource/api/utils/DataSourceUtils.java]:
```suggestion
            logger.info("start register processor: {}", name);
```

File: dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-api/src/main/java/org/apache/dolphinscheduler/plugin/datasource/api/utils/DataSourceUtils.java [hunk 3]
```diff
@@ -17,43 +17,55 @@
...
+            if (!names.add(name)) {
+                throw new IllegalStateException(format("Duplicate datasource plugins named '%s'", name));
+            }
+            loadDatasourceClient(factory);
+
+            logger.info("done register processor: " + name);
```
Comment (caishunfeng) [inline dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-api/src/main/java/org/apache/dolphinscheduler/plugin/datasource/api/utils/DataSourceUtils.java]:
```suggestion
            logger.info("done register processor: {}", name);
```

File: dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-api/src/main/java/org/apache/dolphinscheduler/plugin/datasource/api/utils/DataSourceUtils.java [hunk 4]
```diff
@@ -17,43 +17,55 @@
...
+
+        });
+        int i = 1;
```
Comment (caishunfeng) [inline dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-api/src/main/java/org/apache/dolphinscheduler/plugin/datasource/api/utils/DataSourceUtils.java]: remove if not use.

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/DataSourceController.java [hunk 5]
```diff
@@ -92,7 +90,8 @@ public class DataSourceController extends BaseController {
     @AccessLogAnnotation(ignoreRequestArgs = "loginUser")
     public Result createDataSource(@ApiIgnore @RequestAttribute(value = Constants.SESSION_USER) User loginUser,
                                    @ApiParam(name = "dataSourceParam", value = "DATA_SOURCE_PARAM", required = true)
-                                   @RequestBody BaseDataSourceParamDTO dataSourceParam) {
+                                   @RequestBody String jsonStr) {
```
Comment (caishunfeng) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/DataSourceController.java:94]: It's better to provide json example.

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/ApiApplicationServer.java (hunk 1 above)
Comment (Tianqi-Dotes) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/ApiApplicationServer.java]: done change

File: dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-api/src/main/java/org/apache/dolphinscheduler/plugin/datasource/api/utils/DataSourceUtils.java (hunk 2 above)
Comment (Tianqi-Dotes) [reply dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-api/src/main/java/org/apache/dolphinscheduler/plugin/datasource/api/utils/DataSourceUtils.java]: done

File: dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-api/src/main/java/org/apache/dolphinscheduler/plugin/datasource/api/utils/DataSourceUtils.java (hunk 3 above)
Comment (Tianqi-Dotes) [reply dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-api/src/main/java/org/apache/dolphinscheduler/plugin/datasource/api/utils/DataSourceUtils.java]: done

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/DataSourceController.java (hunk 5 above)
Comment (Tianqi-Dotes) [reply dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/controller/DataSourceController.java:94]: done

**Review by caishunfeng — APPROVED** [review]
LGTM

**Pull request merged**